### DISCUSSION TITLE
Saving Hamiltonian matrix as numpy file

### DISCUSSION THREAD

**Discussion by Brian-Khor** [QuSpin/QuSpin#640] [Q&A]
Dear Admin for QuSpin,

I'm new to QuSpin, and I would like to save the Hamiltonian matrix generated by quspin.operators.hamiltonian as text and numpy files to gain some intuition about the Hamiltonian I'm working with. I realise that the hamiltonian object comes with other attributes and many methods and it is not merely a numpy matrix. Can I know how can I do that once I created the Hamiltonian from the basis?

Best,
Brian.

**Comment by mgbukov** [comment]
Hi, you can use `H.todense()` to turn a `hamiltonian` object into a numpy
array. From then on, you should follow the procedure to store a numpy
array..

On Wed, Aug 23, 2023 at 7:34 PM Brian Jia Jiunn Khor <
***@***.***> wrote:

> Dear Admin for QuSpin,
>
> I'm new to QuSpin, and I would like to save the Hamiltonian matrix
> generated by quspin.operators.hamiltonian as text and numpy files to gain
> some intuition about the Hamiltonian I'm working with. I realise that the
> hamiltonian object comes with other attributes and many methods and it is
> not merely a numpy matrix. Can I know how can I do that once I created the
> Hamiltonian from the basis?
>
> Best,
> Brian.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/QuSpin/QuSpin/discussions/640>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ADHSUBTE36BIPQFYH3VDH73XWY5KHANCNFSM6AAAAAA332DIX4>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Marin Bukov, PhD
Research Group Leader
Max Planck Institute for the Physics of Complex Systems
Noethnitzer Str. 38
Dresden 01187, Germany

url: https://www.pks.mpg.de/nqd
<http://quantum-dynamics.phys.uni-sofia.bg/>
